Migrate OLM bundle image CI job from Gitlab to Github

shivakunv opened the pull request.

shivakunv commented on .github/workflows/ci.yaml:
```diff
@@ -208,109 +209,161 @@ jobs:
+  build-gpu-operator-bundle-image:
+    # needs: [go-check, go-test, go-build]
+    # SHIVA could be only branch=release 
+    if: github.ref_type == 'branch' # # Runs only if the event is from a branch commit
```
on main

shivakunv commented on Makefile:
```diff
@@ -140,7 +140,7 @@ build-bundle-image:
 	--build-arg VERSION=$(VERSION) \
 	--build-arg DEFAULT_CHANNEL=$(DEFAULT_CHANNEL) \
-	--build-arg GIT_COMMIT=$(GIT_COMMIT) \
+	--build-arg GIT_COMMIT=$(GITHUB_SHA) \
```
check if it is needed or not

tariq1890 commented on .github/workflows/ci.yaml:
```diff
@@ -210,7 +210,37 @@ jobs:
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build bundle-image
+        env:
+          BUNDLE_IMAGE: "ghcr.io/nvidia/gpu-operator/gpu-operator-bundle:${{ env.BRANCH_NAME }}-latest"
```
You need this field as well
```suggestion
          BUNDLE_IMAGE: "ghcr.io/nvidia/gpu-operator/gpu-operator-bundle:${{ env.BRANCH_NAME }}-latest"
          VERSION: ""
```

cdesiniotis commented on .github/workflows/ci.yaml:
```diff
@@ -210,7 +210,37 @@ jobs:
+        id: vars
+        run: |
+          echo "BRANCH_NAME=${GITHUB_REF_NAME}" >> $GITHUB_ENV
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+          image: tonistiigi/binfmt:master
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
```
We shouldn't need QEMU or buildx since the bundle image is not multi-arch.

shivakunv commented:
PTAL : 
added VERSION 
removed Qemu and buildx.

passed pipeline 
https://github.com/NVIDIA/gpu-operator/actions/runs/13170036860/job/36758562744

tariq1890 commented on .github/workflows/ci.yaml:
```diff
@@ -210,7 +210,32 @@ jobs:
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build bundle-image
+        env:
+          BUNDLE_IMAGE: "ghcr.io/nvidia/gpu-operator/gpu-operator-bundle:${{ env.BRANCH_NAME }}-latest"
```
Why not this? If this works, then we don't need L219-222
```suggestion
          BUNDLE_IMAGE: "ghcr.io/nvidia/gpu-operator/gpu-operator-bundle:${{ github.ref_name }}-latest"
```

shivakunv replied:
@tariq1890 
This approach is followed for future branch modifications.
If we consider both the release(feature) and main branches, we need to add one 'or' condition.
We don't need to change the code in the run

if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/release'}}

shivakunv replied:
used GITHUB_REF_NAME

tariq1890 commented on .github/workflows/ci.yaml:
```diff
@@ -210,7 +210,32 @@ jobs:
 
-  ### e2e tests ###
+  build-gpu-operator-bundle-image:
+    if: github.ref == 'refs/heads/main' # Runs only if the event is from a main branch commit
```
```suggestion
    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }} # Runs only if the event is a push to the main branch
```

shivakunv replied:
you mean main here ?

tariq1890 commented on .github/workflows/ci.yaml:
```diff
@@ -210,7 +210,32 @@ jobs:
-  ### e2e tests ###
+  build-gpu-operator-bundle-image:
+    if: github.ref == 'refs/heads/main' # Runs only if the event is from a main branch commit
+    runs-on: ubuntu-latest
```
We only want the OLM bundle to be built and pushed after the e2e tests pass as it is going to be consumed by external teams
```suggestion
    needs: [e2e-tests-containerd, e2e-tests-nvidiadriver]
    runs-on: ubuntu-latest
```

tariq1890 commented on .github/workflows/ci.yaml:
```diff
@@ -210,7 +210,32 @@ jobs:
           make build-${{ matrix.dist }}
 
-  ### e2e tests ###
+  build-gpu-operator-bundle-image:
```
Let's place this whole block after the e2e-tests please

tariq1890 replied:
```suggestion
  release-gpu-operator-bundle-image:
```

shivakunv commented on .github/workflows/ci.yaml:
```diff
@@ -314,3 +314,29 @@ jobs:
           retention-days: 15
+
+  push-gpu-operator-bundle-image:
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} # Runs only if the event is a push to the main branch
```
This approach is followed for future branch modifications.
If we consider both the release(feature) and main branches, we need to add one 'or' condition.
We don't need to change the code in the run

if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/release'}}

tariq1890 replied:
Right now, we only want to trigger this on pushes to main.
